base/bucket_n1ql_test.go

```diff
@@ -575,3 +574,37 @@ func tearDownTestIndex(bucket *CouchbaseBucketGoCB, indexName string) (err error
 	}
 
 }
+
+func TestWaitForBucketExistence(t *testing.T) {
+	if UnitTestUrlIsWalrus() {
+		t.Skip("This test only works against Couchbase Server")
+	}
+
+	testBucket := GetTestBucket(t)
+	defer testBucket.Close()
+	bucket, ok := testBucket.Bucket.(*CouchbaseBucketGoCB)
```

Comment:
This will do any conversions from a logging or test bucket into a gocb bucket

```suggestion
	bucket, ok := AsGoCBBucket(testBucket)
```

Reply:
This change has been committed to the latest version.